File: CPP/Program/Motorcycle.cpp
//menggunakan pragma once agar header file hanya di-include satu kali
#pragma once

//memasukan library yang digunakan, termasuk kelas induk (kelas Vehicle) dari kelas Motorcycle
#include <iostream>
#include <string>
#include "Vehicle.cpp"

//deklarasi namespace
using namespace std;

//Kelas Motorcycle yang digunakan untuk merepresentasikan data Motorcycle, Motorcycle merupakan kelas turunan dari kelas Vehicle
class Motorcycle : public Vehicle{
    private:
        //atribut yang digunakan pada kelas Motorcycle terdiri dari jenis motor dan kapasitas tangki
        string jenis_motor;
        int kapasitas_tangki;

    public:
        //constructor
        Motorcycle(){

        }

        //constructor: overloading
        Motorcycle(string kode, string plat_nomor, string merk, int tahun_produksi, string warna, string jenis_motor, int kapasitas_tangki)
        : Vehicle(kode, plat_nomor, merk, tahun_produksi, warna), jenis_motor(jenis_motor), kapasitas_tangki(kapasitas_tangki) {}

        //menampilkan atribut motorcycle
        void displayInfo() const override {
            Vehicle::displayInfo();
            cout << "Jenis Motor        : " << jenis_motor << endl;
            cout << "Kapasitas Tangki   : " << kapasitas_tangki << " liter" << endl;
        }

        //method set untuk set setiap nilai atribut pada kelas Motorcycle
        void setJenisMotor(string jenis_motor){
            this->jenis_motor = jenis_motor;
        }

        void setKapasitasTangki(int kapasitas_tangki){
            this->kapasitas_tangki = kapasitas_tangki;
        }

        //method get untuk mendapatkan nilai setiap atribut pada kelas Motorcycle
        string getJenisMotor(){
            return jenis_motor;
        }

        int getKapasitasTangki(){
            return kapasitas_tangki;
        }

        //destructor
        ~Motorcycle(){

        }
};

File: Python/Program/Motorcycle.py
# import class file
from Vehicle import Vehicle

# deklarasi class
class Motorcycle(Vehicle):
    # constructor
    def __init__(self, kode_kendaraan, plat_nomor, merk, tahun_produksi, warna, jenis_motor, kapasitas_tangki):
        super().__init__(kode_kendaraan, plat_nomor, merk, tahun_produksi, warna)
        self.jenis_motor = jenis_motor
        self.kapasitas_tangki = kapasitas_tangki

    # menampilkan atribut car
    def display_info(self):
        super().display_info()
        print("Jenis Motor        :", self.jenis_motor)
        print("Kapasitas Tangki   :", self.kapasitas_tangki)
        
    # method setter untuk set nilai setiap atribut pada kelas Motorcycle        
    # set jenis motor
    def set_jenis_motor(self, jenis_motor):
        self.jenis_motor = jenis_motor
        
    # set kapasitas tangki
    def set_kapasitas_tangki(self, kapasitas_tangki):
        self.kapasitas_tangki = kapasitas_tangki
        
    # method getter untuk mendapatkan nilai setiap atribut pada kelas Motorcycle
    # get jenis motor
    def get_jenis_motor(self):
        return self.jenis_motor
    
    # get kapasitas tangki
    def get_kapasitas_tangki(self):
        return self.kapasitas_tangki